controllers/project_fetch.go

```diff
@@ -0,0 +1,126 @@
+package controllers
+
+import (
+	"fmt"
+	"kwoc-backend/middleware"
+	"kwoc-backend/models"
+	"kwoc-backend/utils"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"gorm.io/gorm"
+)
+
+type FetchProjMentor struct {
+	Name     string `json:"name"`
+	Username string `json:"username"`
+}
+type FetchProjProject struct {
+	Id              uint            `json:"id"`
+	Name            string          `json:"name"`
+	Desc            string          `json:"desc"`
+	Tags            string          `json:"tags"`
+	RepoLink        string          `json:"repo_link"`
+	ComChannel      string          `json:"com_channel"`
+	ReadmeURL       string          `json:"readme_url"`
+	Mentor          FetchProjMentor `json:"mentor"`
+	SecondaryMentor FetchProjMentor `json:"secondary_mentor"`
+}
+
+type FetchAllProjRes []FetchProjProject
+
+func newFetchProjMentor(mentor *models.Mentor) FetchProjMentor {
+	return FetchProjMentor{
+		Name:     mentor.Name,
+		Username: mentor.Username,
+	}
+}
+func newFetchProjProject(project *models.Project) FetchProjProject {
+	return FetchProjProject{
+		Id:              project.ID,
+		Name:            project.Name,
+		Desc:            project.Desc,
+		Tags:            project.Tags,
+		RepoLink:        project.RepoLink,
+		ComChannel:      project.ComChannel,
+		ReadmeURL:       project.README,
+		Mentor:          newFetchProjMentor(&project.Mentor),
+		SecondaryMentor: newFetchProjMentor(&project.SecondaryMentor),
+	}
+}
+
+func FetchAllProjects(w http.ResponseWriter, r *http.Request) {
```

Comment:
```suggestion
func FetchAllMentorProjects(w http.ResponseWriter, r *http.Request) {
```

lets keep names such that it avoids any confusion. `FetchAllProjects` made me think once whether it fetches all projects of one mentor or all the mentors